src/parse.y

```diff
@@ -1265,24 +1282,12 @@ type:
 				$$ = new VectorType({AdoptRef{}, $3});
 				}
 
-	|	TOK_FUNCTION func_params
-				{
-				set_location(@1, @2);
-				$$ = $2;
-				}
-
 	|	TOK_EVENT '(' formal_args ')'
```

Comment:
Any reason you kept `event`?

The following works and prints `type`, but it doesn't work for hook and function types, so my thinking is that it would be more consistent if it wouldn't work for `event` either (or make it work for all of them), just due to their close relationships, even if technically possible.

```
event zeek_init()
        {
        local et = event();
        print type_name(et);
        local etc = event(c: count);
        print type_name(etc);

        # local ht = hook(c: count);
        # local ft = function(c: count): bool;
        }
```

Reply:
Reason was `simple_type` was all of the types that didn't make a conflict within `expr` - but I moved it out because having some reason beyond "it's easy to parse" makes sense :)